src/Features/VisualBasic/Portable/IntroduceVariable/VisualBasicIntroduceVariableService.Rewriter.vb

```diff
@@ -30,6 +30,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.IntroduceVariable
                 Return MyBase.Visit(node)
             End Function
 
+            Public Overrides Function VisitMemberAccessExpression(node As MemberAccessExpressionSyntax) As SyntaxNode
+                ' Ignore name node in member access expression, since we can't replace it with the new variable.
+                ' For example:
+                '
+                '   Class C
+                '       Dim c As C
+                '       Sub Foo()
+                '           Dim y = [|c|].c.c
+                '       End Sub
+                '   End Class
+
+                Dim newExpression = DirectCast(Visit(node.Expression), ExpressionSyntax)
+                If node.Expression IsNot newExpression Then
+                    Return SyntaxFactory.MemberAccessExpression(node.Kind, newExpression, node.OperatorToken, node.Name)
+                Else
+                    Return node
+                End If
```

Comment:
the lack of symettry here withteh C# version is suprising.

Reply:
You mean `Update` vs. `SyntaxFactory`?

Reply:
yup.  i would have expected teh c#/vb sides to be nearly identical.  is tehre a reason they aren't?

Reply:
Here's the code for `VisualBasicSyntaxRewriter`, it's using internal types all over the place.

```vb
        Public Overrides Function VisitMemberAccessExpression(ByVal node As MemberAccessExpressionSyntax) As SyntaxNode
            Dim anyChanges As Boolean = False

            Dim newExpression = DirectCast(Visit(node.Expression), ExpressionSyntax)
            If node.Expression IsNot newExpression Then anyChanges = True
            Dim newOperatorToken = DirectCast(VisitToken(node.OperatorToken).Node, InternalSyntax.PunctuationSyntax)
            If node.OperatorToken.Node IsNot newOperatorToken Then anyChanges = True
            Dim newName = DirectCast(Visit(node.Name), SimpleNameSyntax)
            If node.Name IsNot newName Then anyChanges = True

            If anyChanges Then
                Return New MemberAccessExpressionSyntax(node.Kind, node.Green.GetDiagnostics, node.Green.GetAnnotations, newExpression, newOperatorToken, newName)
            Else
                Return node
            End If
        End Function
```